src/main/java/Obstacle.java

```diff
@@ -0,0 +1,51 @@
+public class Obstacle {
```

Comment:
Not much of it's functionality is implemented inside this class. clearObstacle cold for example check if the correct item is being used with the itemNeeded member instead of doing this outside the class.